sky/engine/core/rendering/RenderParagraph.cpp

```diff
@@ -880,7 +881,8 @@ void RenderParagraph::layoutRunsAndFloatsInRange(LineLayoutState& layoutState,
         endOfLine = lineBreaker.nextLineBreak(resolver, layoutState.lineInfo(), renderTextInfo,
             lastFloatFromPreviousLine, wordMeasurements);
         renderTextInfo.m_lineBreakIterator.resetPriorContext();
-        if (resolver.position().atEnd()) {
+        m_didExceedMaxLines = layoutState.lineInfo().lineIndex() > styleToUse->maxLines() && !layoutState.lineInfo().isEmpty();
```

Comment:
I'd probably set this back to false somewhere early in layout.  Maybe at the top of this function.